jenkins/pipeline-parallel-mtr.groovy

```diff
@@ -611,7 +617,34 @@ pipeline {
                         git branch: JENKINS_SCRIPTS_BRANCH, url: JENKINS_SCRIPTS_REPO
 
                         checkoutSources()
+                        
+                        // Download ccache using shared library
+                        ccacheDownload([
+                            awsCredentialsId: AWS_CREDENTIALS_ID,
+                            s3Bucket: S3_ROOT_DIR + '/',
+                            workspace: env.WORKSPACE,
+                            branch: env.BRANCH,
+                            dockerOs: env.DOCKER_OS,
+                            cmakeBuildType: env.CMAKE_BUILD_TYPE,
+                            compiler: env.COMPILER,
```

Comment:
The `COMPILER` is undefined by default in Jenkins.
To get best cache hits we need a full compiler version (e.g. `clang-18.1.8` or `gcc-13.3.0`)
In Azure Pipelines we use: https://github.com/percona/percona-server/blob/8.0/azure-pipelines.yml#L363-L367

Reply:
Maybe with Jenkins it's not a big deal because we will change a compiler e.g. when docker images will be updated. With Azure we test many compilers with the same docker image.

Reply:
Completely agree with @inikep here. Object names under which we upload ccache tarballs on AWS S3 should include at least the following 3 components:
1. \<mysql_version> (for instance, `8.0.42` or `8.4.5`)
2. \<toolset> (for instance, `clang-18.1.8` or `gcc-13.3.0`)
3. \<build_configuration> (`Debug` or `RelWithDebInfo`). Ideally ASan (`Debug` + `-DWITH_ASAN=ON`) and Valgrind (`Debug` + `-DWITH_Valgrind=ON`) should also have unique \<build_configuration> labels (e.g. `ASasn` and `Valgrind`).